packages/lodestar/src/chain/attestation/processor.ts

```diff
@@ -41,10 +46,77 @@ export class AttestationProcessor {
  */
 export async function processAttestationJob(modules: AttestationProcessorModules, job: IAttestationJob): Promise<void> {
   try {
-    validateAttestation({...modules, job});
+    // validate attestation in the forkchoice
     await processAttestation({...modules, job});
   } catch (e) {
-    // above functions only throw AttestationError
-    modules.emitter.emit(ChainEvent.errorAttestation, e);
+    // above functions ForkChoice attestation error, we have to map it to AttestationError
+    modules.emitter.emit(ChainEvent.errorAttestation, mapAttestationError(e, job) || e);
   }
 }
+
+/**
+ * Map ForkChoice attestation error to lodestar version.
+ */
+function mapAttestationError(e: Error, job: IAttestationJob): AttestationError | null {
+  if (e instanceof ForkChoiceError && e.type.code === ForkChoiceErrorCode.INVALID_ATTESTATION) {
+    const attError = e.type.err;
+    switch (attError.code) {
+      case InvalidAttestationCode.EMPTY_AGGREGATION_BITFIELD:
+        return new AttestationError({
+          code: AttestationErrorCode.EMPTY_AGGREGATION_BITFIELD,
```

Comment:
It seems that the errors map 1 to 1?
Can we have an implementation that looks like this (may need some tweaking):
```ts
if (e instanceof ForkChoiceError && e.type.code === ForkChoiceErrorCode.INVALID_ATTESTATION) {
    const attError = e.type.err;
    (attError as AttestationError).job = job;
    return attError as AttestationError;
  }
```
This way we can retain the right stack trace